### ISSUE TITLE
BUG: coding UTF8

### ISSUE THREAD

**Issue by pburdet** [hyperspy/hyperspy#549]
It seems that pull request #526 introduces a bug. The list of string in metadata are saved (or load) as list of ``[u'a', u'b']`` instead of ``['a', 'b']``, see following examples:

```python

>>> s = signals.Spectrum(range(10))
>>> s.metadata.set_item('Sample.elements', ['a', 'b'])
>>> print isinstance(a, str)
>>> s.save('test', overwrite=True)
>>> s = load('test.hdf5')
>>> a = s.metadata.Sample.elements[0]
>>> print isinstance(a, str)
True
False

```

**Comment by francisco-dlp** [comment]
I agree that this can be considered a bug, but I'm not sure if we should fix it, specially given that in python3 all the strings will be unicode. Do you have any arguments in favour of fixing this?

**Comment by pburdet** [comment]
I reported it because it breaks the 'eds_model' branch. Here is where it creates problem:

```python
s = utils.example_signals.EDS_SEM_Spectrum()
s.add_lines()
s.save('test', overwrite=True)
s = load('test.hdf5')
m = create_model(s, auto_add_lines=False)
c = components.Gaussian()
c.name = s.metadata.Sample.xray_lines[0]
m.append(c)
m[s.metadata.Sample.xray_lines[0]]

```

**Comment by vidartf** [comment]
Error here is that `Model.__getitem__()` checks for `str`: `if isinstance(value, str):`. If it instead checks for `basestring`, we should be OK (I think equality operators of `str` and `unicode` will handle correctly  them being different types, but not certain).

**Comment by vidartf** [comment]
(Note: `basestring` is the common base of `str` and `unicode`)

**Comment by pburdet** [comment]
Ok it is easy to fix. Maybe, all the `if isinstance(value, str):` should be tested.

**Comment by vidartf** [comment]
A quick suggestion in #556.

**Comment by francisco-dlp** [comment]
Fixed in #557
